language/move-prover/boogie-backend/src/options.rs

```diff
@@ -153,13 +153,19 @@ impl BoogieOptions {
     }
 
     /// Returns command line to call boogie.
-    pub fn get_boogie_command(&self, boogie_file: &str) -> Vec<String> {
+    pub fn get_boogie_command(&self, boogie_file: &str) -> anyhow::Result<Vec<String>> {
         let mut result = if self.use_exp_boogie {
             // This should have a better ux...
             vec![read_env_var("EXP_BOOGIE_EXE")]
         } else {
             vec![self.boogie_exe.clone()]
         };
+
+        // If we don't have a boogie executable, nothing will work
+        if result.iter().all(|path| path.is_empty()) {
```

Comment:
Just a comment that the message might eventually need to be different based on whether the user was expected to supply BOOGIE_EXE or EXP_BOOGIE_EXE.